src/dune_pkg/rev_store.ml

```diff
@@ -684,25 +685,33 @@ module At_rev = struct
       >>| List.cons files
       >>| File.Set.union_all
     in
-    { repo; revision; files }
+    let descendants =
+      let descendants = String.Table.create (File.Set.cardinal files) in
+      File.Set.iter files ~f:(fun file ->
+        let rec loop = function
+          | None -> ()
+          | Some parent ->
+            let key = Path.Local.to_string parent in
+            let descendants_of_parent =
+              String.Table.find_or_add descendants key ~f:(Fun.const File.Set.empty)
+            in
+            let descendants_of_parent = File.Set.add descendants_of_parent file in
+            String.Table.set descendants key descendants_of_parent;
+            loop (Path.Local.parent parent)
+        in
+        loop (File.path file |> Path.Local.parent));
```

Comment:
I'm a bit lost here: you are looking for _descendant_, but you are calling `Path.Local.parent`. What am I missing here? :sweat_smile: